### PR TITLE
Update ethnicity codelist

### PR DESCRIPTION
**Pull request by milanwiedemann**
Add  6 and 16 ethnicity groupings to measures

### CODE REVIEW

File: analysis/study_definition_ethnicity.py [hunk 1]
```diff
@@ -16,45 +16,99 @@
...
+            "Asian or Asian British - Indian": "eth='8'",
+            "Asian or Asian British - Pakistani": "eth='9'",
+            "Asian or Asian British - Bangladeshi": "eth='10'",
+            "Asian or Asian British - Any other Asian background": "eth='11'",
+            "Black or Black British - Caribbean": "eth='12'",
+            "Black or Black British - African": "eth='13'",
+            "Black or Black British - Any other Black background": "eth='14'",
+            "Other Ethnic Groups - Chinese": "eth='15'",
+            "Other Ethnic Groups - Any other ethnic group": "eth='16'",
+        },
```
Comment (ccunningham101) [inline analysis/study_definition_ethnicity.py:67]: The codelist also has "17 - Not stated"
Just checking it's a choice to include that as Unknown?
I extracted them separately but then ended up combining them for my analysis
But extracting them together protects against any potential small number suppression issues

File: analysis/study_definition_ethnicity.py [hunk 2]
```diff
@@ -16,45 +16,99 @@
...
             "Unknown": "DEFAULT",
             "White": "eth='1'",
             "Mixed": "eth='2'",
-            "Asian": "eth='3'",
-            "Black": "eth='4'",
-            "Other": "eth='5'",
+            "Asian or Asian British": "eth='3'",
+            "Black or Black British": "eth='4'",
+            "Chinese or Other Ethnic Groups": "eth='5'",
         },
```
Comment (ccunningham101) [inline analysis/study_definition_ethnicity.py:100]: Same here for 6 - Not stated

File: analysis/study_definition_ethnicity.py [hunk 3]
```diff
@@ -16,45 +16,99 @@
...
         {
             "Unknown": "DEFAULT",
             "White": "eth='1'",
             "Mixed": "eth='2'",
-            "Asian": "eth='3'",
-            "Black": "eth='4'",
-            "Other": "eth='5'",
+            "Asian or Asian British": "eth='3'",
+            "Black or Black British": "eth='4'",
+            "Chinese or Other Ethnic Groups": "eth='5'",
```
Comment (ccunningham101) [inline analysis/study_definition_ethnicity.py:99]: For ethnicity6 you care categorizing the eth variable, which is created using the 16 groupings
So right now the ethnicity6 variable will categorize with 
            "White - British": "eth='1'",
            "White - Irish": "eth='2'",
            "White - Any other White background": "eth='3'",
            "Mixed - White and Black Caribbean": "eth='4'",
            "Mixed - White and Black African": "eth='5'", 
            
I think instead you need to extract a new variable patients.with_these_clinical_events(ethnicity6_codes) and then categorize that 

If you were to run flake8 on this file, you might see that ethnicity6_codes is currently unused

**Review by ccunningham101 — CHANGES_REQUESTED** [review]
You probably want to extract a new variable with the ethnicity6 codelist before categorising ethnicity6

File: analysis/study_definition_ethnicity.py (hunk 1 above)
Comment (milanwiedemann) [reply analysis/study_definition_ethnicity.py:67]: oh yes, 17 is in the about but there's no code, so I assume its just missing? [^0] and I couldnt find code for "17' in your code? [^1]

[^0]:  https://www.opencodelists.org/codelist/opensafely/ethnicity-snomed-0removed/2e641f61/
[^1]: https://github.com/opensafely/antidepressant-prescribing-lda/blob/12e4acca89a73c640dc14c602ef65c4c6640894b/analysis/study_definition_ethnicity.py

File: analysis/study_definition_ethnicity.py (hunk 3 above)
Comment (milanwiedemann) [reply analysis/study_definition_ethnicity.py:99]: ohhh yes! thanks! good spot, will fix

Comment (milanwiedemann) [reply analysis/study_definition_ethnicity.py]: fixed https://github.com/opensafely/blood-pressure-sro/blob/ce74d0c6ff100251931cf5d435e2d09808681a3d/analysis/study_definition_ethnicity.py#L92-L131

File: analysis/study_definition_ethnicity.py (hunk 1 above)
Comment (ccunningham101) [reply analysis/study_definition_ethnicity.py:67]: Ah good point that there is no code 17 in the csv. I'll ask Colm about that

**Review by ccunningham101 — APPROVED** [review]
Changes look good

**Pull request merged**